File: javabackend/src/main/java/mft/dev/javabackend/dto/update/UtenteUpdateRequestDTO.java
package mft.dev.javabackend.dto.update;

import com.fasterxml.jackson.annotation.JsonProperty;
import mft.dev.javabackend.model.Utente;

public record UtenteUpdateRequestDTO(
        @JsonProperty("nome")
        String nome,
        @JsonProperty("cognome")
        String cognome,
        @JsonProperty("email")
        String email
) {

        public static Utente toUtente(UtenteUpdateRequestDTO dto, Utente utente){
                Utente updatedUtente = utente;

                if (dto.nome() != null) updatedUtente.setNome(dto.nome());
                if (dto.cognome() != null) updatedUtente.setCognome(dto.cognome());
                if (dto.email() != null && !dto.email().isBlank()) updatedUtente.setEmail(dto.email());

                return updatedUtente;
        }
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/dto/update/UtenteUpdateRequestDTO.kt
package mft.dev.kotlinbackend.dto.update

import com.fasterxml.jackson.annotation.JsonProperty
import mft.dev.kotlinbackend.model.Utente

data class UtenteUpdateRequestDTO(
    @JsonProperty("nome")
    val nome: String?,
    @JsonProperty("cognome")
    val cognome: String?,
    @JsonProperty("email")
    val email: String?
) {
    companion object{
        fun toUtente(dto: UtenteUpdateRequestDTO, utente: Utente): Utente{
            dto.nome?.let { utente.nome = it }
            dto.cognome?.let { utente.cognome = it }
            dto.email?.let { utente.email = it}

            return utente
        }
    }
}
